packages/thyra/core/src/support/nonlinear/model_evaluator/client_support/Thyra_DirectionalFiniteDiffCalculator_def.hpp

```diff
@@ -160,6 +160,9 @@ class OutArgsCreator : public StateFuncModelEvaluatorBase<Scalar>
         << ", the space p_space("<<l<<") must be in-core so that they can"
         " act as the domain space for the multi-vector derivative!"
         );
+#else
+      (void)model;
```

Comment:
Casting to `void` doesn't always fix "unused variable" compiler warnings, but it's a start.

Reply:
Hmmm, I've seen it work any time I've tried it, but perhaps that's just for gcc.  Is there a better way?